tests/test_byteutils.nim

```diff
@@ -65,6 +65,9 @@ suite "Byte utils":
     expect(ValueError): discard hexToByteArray[1]("")
     expect(ValueError): discard hexToByteArray[1]("1")
 
+  test "hexToByteArray: Nully value":
+    expect(ValueError): discard hexToByteArray[1]("0x")
```

Comment:
@arnetheduck actually I see the line below ([87](https://github.com/status-im/nim-stew/pull/144/files#diff-c456240aa7918dfdb6accf51ef55fd6032b80fa58d2bc699614ff53d505e24cbR87))

When we see a `0x` we return an empty array. Since we are extracting a single value here ValueError seems correct, but just so you can confirm.

Reply:
the case where "0x" is given for a 1-byte array, `ValueError` should indeed happen - however, when the caller passes in nonsensical indices, an assertion should happen - your change makes everything a value error which is not the original intent of the function

Reply:
keep in mind also that to/from indices are inclusive of their ends in Nim - an "empty" range is thus `0/-1` while `0/0` is a 1-byte range